lib/propmangle.js

```diff
@@ -150,6 +150,11 @@ function mangle_properties(ast, options) {
             if (!(keep_quoted && node.quote))
                 node.key = mangle(node.key);
         }
+        else if (node instanceof AST_ConciseMethod && node.name && node.key.name) {
```

Comment:
So with the following on #2262
```diff
--- a/lib/propmangle.js
+++ b/lib/propmangle.js
@@ -157,9 +157,9 @@ function mangle_properties(ast, options) {
         if (node instanceof AST_ObjectKeyVal) {
             add(node.key);
         }
-        else if (node instanceof AST_ConciseMethod && node.key && node.key.name) {
-            add(node.key.name);
-        }
+//        else if (node instanceof AST_ConciseMethod && node.key && node.key.name) {
+//            add(node.key.name);
+//        }
         else if (node instanceof AST_ObjectProperty) {
             // setter or getter, since KeyVal is handled above
             add(node.key.name);
@@ -177,9 +177,9 @@ function mangle_properties(ast, options) {
         if (node instanceof AST_ObjectKeyVal) {
             node.key = mangle(node.key);
         }
-        else if (node instanceof AST_ConciseMethod && node.key && node.key.name) {
-            node.key.name = mangle(node.key.name);
-        }
+//        else if (node instanceof AST_ConciseMethod && node.key && node.key.name) {
+//            node.key.name = mangle(node.key.name);
+//        }
         else if (node instanceof AST_ObjectProperty) {
             // setter or getter
             node.key.name = mangle(node.key.name);
```
I have managed to pass `npm test` plus:
```js
$ echo 'console.log({ "Quoted"(){}, Unquoted(){}, "Prop":3 });' | bin/uglifyjs -c --mangle-props keep_quoted
console.log({Quoted(){},o(){},Prop:3});
```